File: src/Problems/RandomPermutation/__tests__/RandomizeInPlace.test.ts
import { randomizeInPlace } from '../RandomizeInPlace';

describe('RandomizeInPlace', function() {
  it('should return array with the same length and values', function() {
    const array = [1, 2, 3, 4, 5];
    const result = [...array];

    randomizeInPlace(result);

    expect(result).toHaveLength(array.length);
    expect(result.sort()).toEqual(array.sort());
  });
});

File: src/Problems/RandomPermutation/__tests__/randomize_in_place_test.py
from src.Problems.RandomPermutation.randomize_in_place import randomize_in_place


class TestRandomizeInPlace:
    def test_same_elements_and_length(self):
        array = [1, 2, 3, 4, 5]
        result = array.copy()

        randomize_in_place(result)

        assert len(result) == len(array)
        assert result.sort() == array.sort()
